### PR TITLE
Better isolate fake whoami service

### PR DESCRIPTION
**Pull request by frenchy64**
**Epic** https://github.com/threatgrid/iroh/issues/3985

This PR moves the fake WhoAmI service provided by the unit tests to a TK service. This removes the need for `whoami-helpers/fixture-reset-state`, and creates better test isolation since a fake server is now booted for every app that uses `whoami-helpers/fixture-server`.

A new service `ThreatgridAuthWhoAmIURLService` is introduced that exposes an easily stubbable `get-whoami-url` method. This is so we don't need to stub the entire `threatgrid-auth-service` service just to provide our own whoami url. Since we only know the port of the whoami url at app initialization, we can't use ConfigService to provide the URL, so this method provides just enough dynamic configuration to get by.

Unit tests are included for `get-whoami-url`, but relies on existing tests for integration with other services (namely `threatgrid-auth-service`).

<a name="qa">[§](#qa)</a> QA
============================

No QA is needed.

<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================

```
intern: Better isolate fake whoami service
```

<a name="squashed-commits">[§](#squashed-commits)</a> Squashed Commits
======================================================================

### CODE REVIEW

File: test/ctia/test_helpers/core.clj [hunk 1]
```diff
@@ -249,8 +249,18 @@
                        "ctia.http.port" http-port
                        "ctia.http.show.port" http-port]
        (let [config (build-transformed-init-config)
+             services-map (cond-> (init/default-services-map config)
+                            (#{:threatgrid} (get-in config [:ctia :auth :type]))
+                            ;; dynamic requires can be removed when #'with-properties is phased out or moved
```
Comment (frenchy64) [inline test/ctia/test_helpers/core.clj:254]: Here we take advantage of services being a map from keywords to services to stub what we want.

File: src/ctia/init.clj [hunk 2]
```diff
@@ -37,41 +38,50 @@
...
   [config]
   (let [{auth-service-type :type} (get-in config [:ctia :auth])
         auth-svc
         (case auth-service-type
-          :allow-all allow-all/allow-all-auth-service
-          :threatgrid threatgrid/threatgrid-auth-service
-          :static static-auth/static-auth-service
+          :allow-all {:IAuth allow-all/allow-all-auth-service}
+          :threatgrid {:IAuth threatgrid/threatgrid-auth-service
+                       :ThreatgridAuthWhoAmIURLService threatgrid/threatgrid-auth-whoami-url-service}
```
Comment (frenchy64) [inline src/ctia/init.clj:55]: Returning a map here makes it really easy to stub out services. See comment in test helpers for an example.

File: test/ctia/auth/threatgrid_test.clj [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+            [ctia.store-service-test :refer [store-services]]
+            [clojure.test :refer [deftest is testing]]
+            [puppetlabs.trapperkeeper.app :as app]
+            [puppetlabs.trapperkeeper.testutils.bootstrap :refer [with-app-with-config]])
+  (:import [java.util UUID]))
+
+(defn threatgrid-auth-whoami-url-services
+  "Service map for #'sut/threatgrid-auth-whoami-url-service"
+  []
+  {:ThreatgridAuthWhoAmIURLService sut/threatgrid-auth-whoami-url-service})
```
Comment (frenchy64) [inline test/ctia/auth/threatgrid_test.clj:12]: Returning maps for the `*-services` functions for easy stubbing and more control over merging service maps.

File: test/ctia/auth/threatgrid_test.clj [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+(defn threatgrid-auth-service-map
+  "Service map for #'sut/threatgrid-auth-service"
+  []
+  (merge (threatgrid-auth-whoami-url-service-map)
+         (store-service-map)
+         {:IAuth sut/threatgrid-auth-service}))
+
+(deftest threatgrid-auth-whoami-url-service-test
+  (doseq [url (doto (not-empty
+                      (repeatedly 10
```
Comment (vemv) [inline test/ctia/auth/threatgrid_test.clj:23]: doto and assert seem unnecessary here since the `repeatedly`  construction always returns a valid list

**Review by vemv — APPROVED** [review]
lgtm!

File: src/ctia/init.clj [hunk 5]
```diff
@@ -38,41 +39,50 @@
...
-       http-server-svc/ctia-http-server-service
-       hooks-svc/hooks-service
-       graphql-svc/graphql-service
-       graphql-registry-svc/graphql-named-type-registry-service
-       riemann/riemann-metrics-service
-       jmx/jmx-metrics-service
-       console/console-metrics-service]
+    (merge-with
+      (fn [l r]
+        (throw (ex-info "Conflict!" {:left l :right r})))
```
Comment (yogsototh) [inline src/ctia/init.clj:71]: Maybe

```suggestion
        (throw (ex-info "Service Graph Conflict!" {:left l :right r})))
```

**Review by yogsototh — APPROVED** [review]
Nice!

**Review by ereteog — APPROVED** [review]
LGTM!

**Comment by frenchy64** [comment]
Thanks for the reviews!

**Pull request merged**